CodeStarterCamp_Week4/main.swift

```diff
@@ -8,5 +8,80 @@
 
 import Foundation
 
-print("Hello, World!")
+class BodyCondition {
+    var 상체근력: Int
+    var 하체근력: Int
+    var 근지구력: Int
+    var 피로도: Int
+    
+    init(상체근력: Int, 하체근력: Int, 근지구력: Int, 피로도: Int){
+        self.상체근력 = 상체근력
+        self.하체근력 = 하체근력
+        self.근지구력 = 근지구력
+        self.피로도 = 피로도
+    }
+    
+   
+}
 
+struct Exercise {
+    let name: String
+    let action: (BodyCondition) -> Void
+}
+
+let 윗몸일으키기: Exercise = Exercise(name: "윗몸일으키기") { users in
+    users.상체근력 += Int.random(in: 10...20)
+    users.피로도 += Int.random(in: 10...20)
+}
+
+let 스쿼트: Exercise = Exercise(name: "스쿼트") { users in
+    users.하체근력 += Int.random(in: 20...30)
+    users.피로도 += Int.random(in: 10...20)
+}
+
+let 오래달리기: Exercise = Exercise(name: "오래달리기") { users in
+    users.근지구력 += Int.random(in: 20...30)
+    users.상체근력 += Int.random(in: 5...10)
+    users.하체근력 += Int.random(in: 5...10)
+    users.피로도 += Int.random(in: 20...30)
+}
+
+let 동적휴식: Exercise = Exercise(name: "동적휴식") { users in
+    users.피로도 = Int.random(in: 5...10)
+}
+
+struct Routine{
+    var routineName: String
+    var exercises: [Exercise]
+    
+    func startRoutine(users: BodyCondition){
+        print("""
+        --------------
+        \(routineName)을 시작합니다.
+        """)
+        
+        for actionIndex in 0..<exercises.count{
+            exercises[actionIndex].action(users)
+            print(exercises[actionIndex].name)
+        }
```

Comment:
```suggestion
        for exercise in exercises {
            exercise.action(user)
            print(exercise.name)
        }
```

이렇게 순회해도 될 것 같은데요? ㅎㅎ 바로 접근할 수 있는데, `Index`를 통해 돌아서 접근하신 것 같아 보여요.

Reply:
- 그렇네요! 생각이 조금 많았던 것 같습니다 ㅎㅎ